### PR TITLE
Expose two SubstructUtils functions to SWIG wrappers

### PR DESCRIPTION
**Pull request by ptosco**
- expose two useful functions from SubtructUtils to SWIG wrappers (`getMostSubstitutedCoreMatch`, `sortMatchesByDegreeOfCoreSubstitution`)
- add unit test to exercise the above

### CODE REVIEW

File: Code/JavaWrappers/ROMol.i [hunk 1]
```diff
@@ -289,6 +290,17 @@ void setPreferCoordGen(bool);
     return RDKit::replaceCore(*($self), coreQuery, replaceDummies, labelByIndex);
   };
 
+  /* Methods from SubstructUtils.h */
+  const RDKit::MatchVectType &getMostSubstitutedCoreMatch(const RDKit::ROMol& core,
+    const std::vector<RDKit::MatchVectType>& matches) {
+    return RDKit::getMostSubstitutedCoreMatch(*($self), core, matches);
```
Comment (greglandrum) [inline Code/JavaWrappers/ROMol.i:296]: I love it when I run across stuff in the public API that I have completely forgotten about. 🤷

**Review by greglandrum — APPROVED** [review]
Definitely useful functionality... if only I knew it was there. ;-)

**Comment by ptosco** [comment]
@greglandrum You didn't know because IIRC it was me adding that code back in time when I added support for the `allowRGroups` flag to `generateDepictionMatching2DStructure` :)

**Comment by greglandrum** [comment]
> @greglandrum You didn't know because IIRC it was me adding that code back in time when I added support for the `allowRGroups` flag to `generateDepictionMatching2DStructure` :)

Yeah, but theoretically I do review these things. :-)

**Pull request merged**